src/strongbox/core.clj

```diff
@@ -862,16 +863,31 @@
   (let [xf (filter #(= name (:name %)))]
     (into [] xf db)))
 
+(defn-spec db-search-sampling? boolean?
+  "returns `true` if a database search should return a random sample.
+  essentially, if nothing has been searched for and no filters have been set, we should take a random
+  sample of the selected catalogue UNLESS something has explicitly flipped the `:sample?` boolean."
+  [search-state map?]
+  (let [filter-by (-> search-state :filter-by)]
+    ;; todo: could we just compare this to the default empty search state ...?
+    (and (:sample? search-state) ;; true by default, set to false to short circuit this logic.
```

Comment:
this is the only difference between `jfx/db-search-sampling?` - can the two be reconciled?